x-pack/plugins/security_solution/public/value_list/components/upload_list_item.tsx

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { EuiButton, EuiFilePicker, EuiToolTip } from '@elastic/eui';
+import { css } from '@emotion/css';
+import { useImportList, useInvalidateListItemQuery } from '@kbn/securitysolution-list-hooks';
+import type { Type } from '@kbn/securitysolution-io-ts-list-types';
+import { useKibana } from '../../common/lib/kibana';
+import { useAppToasts } from '../../common/hooks/use_app_toasts';
+import {
+  UPLOAD_LIST_ITEM,
+  UPLOAD_FILE_PICKER_INITAL_PROMT_TEXT,
+  UPLOAD_TOOLTIP,
+  FAILED_TO_UPLOAD_LIST_ITEM_TITLE,
+  FAILED_TO_UPLOAD_LIST_ITEM,
+  SUCCESSFULY_UPLOAD_LIST_ITEMS,
+} from '../translations';
+
+const validFileTypes = ['text/csv', 'text/plain'];
+
+const toastOptions = {
+  toastLifeTimeMs: 5000,
+};
+
+const uploadStyle = css`
+  min-width: 300px;
+`;
+
+export const UploadListItem = ({ listId, type }: { listId: string; type: Type }) => {
+  const [file, setFile] = useState<File | null>(null);
+  const { http } = useKibana().services;
+  const ctrl = useRef(new AbortController());
+  const { start: importList, ...importState } = useImportList();
+  const invalidateListItemQuery = useInvalidateListItemQuery();
+  const { addSuccess, addError } = useAppToasts();
+  const handleFileChange = useCallback((files: FileList | null) => {
+    setFile(files?.item(0) ?? null);
+  }, []);
+
+  const handleImport = useCallback(() => {
+    if (!importState.loading && file) {
+      ctrl.current = new AbortController();
+      importList({
+        file,
+        listId,
+        http,
+        signal: ctrl.current.signal,
+        type,
+        refresh: 'true',
+      });
+    }
+  }, [importState.loading, file, importList, http, type, listId]);
+
+  const fileIsValid = !file || validFileTypes.some((fileType) => file.type === fileType);
```

Comment:
Nit: since `validFileTypes` is already outside this component, might as well move this to a helper